hummingbot/connector/exchange/btc_markets/btc_markets_exchange.py

```diff
@@ -0,0 +1,966 @@
+            self._account_available_balances[asset_name] = Decimal(str(account["available"]))
+            self._account_balances[asset_name] = Decimal(str(account["balance"]))
+            # self._account_locked_balances[asset_name] = Decimal(str(account["locked"]))
+            remote_asset_names.add(asset_name)
+
+        asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+        for asset_name in asset_names_to_remove:
+            del self._account_available_balances[asset_name]
+            del self._account_balances[asset_name]
+
+        # as self._real_time_balance_update = false
+        # take `in_flight_orders` snapshot during `_update_balances` (see #2224)
+        self._in_flight_orders_snapshot = {k: copy.copy(v) for k, v in self._in_flight_orders.items()}
+        self._in_flight_orders_snapshot_timestamp = btc_markets_utils.get_ms_timestamp()
+
+    async def _update_order_status(self):
+        """
+        Calls REST API to get status update for each in-flight order.
+        """
+
+        last_tick = int(self._last_poll_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+        current_tick = int(self.current_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+
+        if current_tick > last_tick and len(self._in_flight_orders) > 0:
+            tracked_orders = list(self._in_flight_orders.values())
+            tasks = []
+            for tracked_order in tracked_orders:
+                order_id = await tracked_order.get_exchange_order_id()
+                tasks.append(self._api_request("GET",
+                                               f"{Constants.ORDERS_URL}/{order_id}",
+                                               {},
+                                               # Constants.ORDERS_URL,
+                                               # {"id": order_id},
+                                               True))
+            self.logger().debug(f"Polling for order status updates of {len(tasks)} orders.")
+            responses = await safe_gather(*tasks, return_exceptions=True)
+            for response in responses:
+                if isinstance(response, Exception):
+                    raise response
+                if "status" not in response:
+                    self.logger().info(f"_update_order_status result not in resp: {response}")
+                    continue
+
+                if "trades" in response:
+                    await self._process_trade_message(response)
+
+                await self._process_order_message(response)
+
+    async def _process_order_message(self, order_msg: Dict[str, Any]):
+        """
+        Updates in-flight order and triggers cancellation or failure event if needed.
+        :param order_msg: The order response from either REST or WebSocket but WS doesn't return clientOrderId on
+        orderChange events. https://api.btcmarkets.net/doc/v3#section/Order-life-cycle-events
+        """
+
+        if "clientOrderId" in order_msg:
```

Comment:
The API documentation does not specify that the `clientOrderId` is included in the order events sent using the websocket connection. In case it is really included and it is an error in the documentation, please review the current implementation. It does not make sense to process an event twice by finding the in flight order using both the client order id and then the order id.